doc/releases/changelog-dev.md

```diff
@@ -56,6 +56,7 @@
 * Support for sample-based measurements has been added to the `DefaultQubit2` device.
   [(#4105)](https://github.com/PennyLaneAI/pennylane/pull/4105)
   [(#4114)](https://github.com/PennyLaneAI/pennylane/pull/4114)
+  [(#4120)](https://github.com/PennyLaneAI/pennylane/pull/4120)
```

Comment:
Minor: maybe I would describe the change for this specific PR as well

Reply:
Added a separate entry 👍